SalesforceCommerceCloud/pwa-kit
# @W-15490276@ Better refresh token error logging for V3

Related to #1829 

There are a few ways in which a call to SLAS to consume a refresh token can fail. For example, the client id the refresh token is associated with might be different from the client id included with the request.

Before this change, the error logging for a failed refresh token call only triggered when SLAS returned the error message 'invalid refresh token'. 

This change opens up the logging so that any failed refresh token call error message is logged.

This change also updates new guest login to also log error messages.

Testing the changes:
Refresh login (note: the refresh login request happens client side only):
1. Start the app with these changes locally (this will start the app against zzrf-001)
2. Delete the access token in your local store
3. On a different browser window, open up https://test.phased-launch-testing.com/ and grab the refresh token
4. Copy the refresh token from test.phased-launch-testing into the cc-nx-g cookie on the tab that is open on zzrf-001
5. Refresh the tab on zzrf-001
6. See in the browser console that there is a log entry for `commerce-sdk-react ERROR invalid organizationId`
7. Verify that your refresh token was replaced with a new one that is usable for zzrf-001

New guest login:
1. In default.js, set an invalid client ID
2. Start the app (make sure you don't have an existing refresh token or access token)
3. In server logs, see that there is a log entry for 400 Bad Request with a descriptive SLAS error message (ie. client_id must be 36 chars)
4. In browser console, see the same message.
